### PR TITLE
[REVPI-2581] Adjust the install debs script to generate up to date image

### PR DESCRIPTION
**Pull request by zhan-kunbus**
1. for "minimized" image
2. for dt-blob.bin
3. add sync before cleanup mount
4. add ability of add files

### CODE REVIEW

File: install_debs_into_image.sh [hunk 1]
```diff
@@ -66,6 +66,7 @@ LOOPDEVICE=$(losetup -f)
 CONFIGTXT="$IMAGEDIR/boot/config.txt"
 
 cleanup_umount() {
+	sync
```
Comment (iluminat23) [inline install_debs_into_image.sh:69]: Why is this needed?

File: install_debs_into_image.sh [hunk 2]
```diff
@@ -172,6 +172,9 @@ find "$IMAGEDIR/etc/ssh" -name "ssh_host_*_key*" -delete
 # restore ld.so.preload
 mv "$IMAGEDIR/etc/ld.so.preload.bak" "$IMAGEDIR/etc/ld.so.preload"
 
+# after package raspberrypi-kernel installed, install revpi-dt-blob.dtbo as default dt-blob
+install -T "$IMAGEDIR/boot/overlays/revpi-dt-blob.dtbo" "$IMAGEDIR/boot/dt-blob.bin"
```
Comment (iluminat23) [inline install_debs_into_image.sh:186]: Why do we need this in the install_debs_to_image script?

File: install_debs_into_image.sh (hunk 1 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:69]: I see this sometimes:
`[2022-10-14 08:40:12]  install file:/lib/firmware/renesas_usb_fw.mem
[2022-10-14 08:40:13]  fsck.fat 4.2 (2021-01-31)
[2022-10-14 08:40:14]  There are differences between boot sector and its backup.
[2022-10-14 08:40:14]  This is mostly harmless. Differences: (offset:original/backup)
[2022-10-14 08:40:14]    65:01/00
[2022-10-14 08:40:14]    Not automatically fixing this.
[2022-10-14 08:40:14]  Dirty bit is set. Fs was not properly unmounted and some data may be corrupt.
[2022-10-14 08:40:14]   Automatically removing dirty bit.
[2022-10-14 08:40:14]  
[2022-10-14 08:40:14]  *** Filesystem was changed ***
[2022-10-14 08:40:14]  Writing changes.
[2022-10-14 08:40:14]  /dev/loop2p1: 322 files, 85960/516190 clusters
[2022-10-14 08:40:14]  /dev/loop2: [2065]:39512 (/mnt/sdb/1/2022-07-28-revpi-buster-lite.img)`

File: install_debs_into_image.sh (hunk 2 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:186]: I think it has the same reason as it in customize_image.sh

Comment (iluminat23) [reply install_debs_into_image.sh:186]: But it is not the same use case as for the customize_image.sh. The script is only intended to add packages to an image. And modifying the dt-blob.bin is not something the script should do.

File: install_debs_into_image.sh (hunk 1 above)
Comment (iluminat23) [reply install_debs_into_image.sh:69]: If this change should be considered, please add a description on the circumstances this can happen.

File: install_debs_into_image.sh [hunk 3]
```diff
@@ -165,6 +165,16 @@ if [ "$(/bin/ls "$BAKERYDIR/debs-to-install/"*.deb 2>/dev/null)" ] ; then
 	chroot "$IMAGEDIR" sh -c "dpkg -i /tmp/debs-to-install/*.deb"
 fi
 
+# install local files
+LIST_INS="$BAKERYDIR/files-to-install/list.ins"
```
Comment (iluminat23) [inline install_debs_into_image.sh:169]: I'm not sure about this change. Also there is not description about this feature.

File: install_debs_into_image.sh (hunk 2 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:186]: when the kernel package gets updated, the /boot/dt-blob.dtbo should be replace with new revpi-dt-blob.dtbo, is it?

File: install_debs_into_image.sh [hunk 4]
```diff
@@ -2,7 +2,8 @@
 # Customize packages of an existing image for Revolution Pi
 
 usage () {
-	echo 'Usage: install_debs_into_image.sh [-h, --help] <image>
+	echo 'Usage: install_debs_into_image.sh [-m, --minimize |-h, --help] <image>
```
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply install_debs_into_image.sh:5]: the (min-)debs-to-download  are still used in script.

[the remark this answers is not in the document]
Comment (iluminat23) [reply install_debs_into_image.sh:5]: This makes the script more complicated then needed. Just place the packages you want to install in the directory.

File: install_debs_into_image.sh (hunk 2 above)
Comment (iluminat23) [reply install_debs_into_image.sh:186]: No, as this script has no knowledge about what has changed it might override stuff it is not intended to override. The install_debs_to_image script should I guess only install debs...

Comment (zhan-kunbus) [reply install_debs_into_image.sh:186]: It was added because of https://github.com/RevolutionPi/imagebakery/pull/75#discussion_r945642016

File: install_debs_into_image.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply install_debs_into_image.sh:5]: It was added because of https://github.com/RevolutionPi/imagebakery/pull/75#discussion_r945642016

[the remark this answers is not in the document]
Comment (iluminat23) [reply install_debs_into_image.sh:5]: No, the minimize option makes THIS script more complicated then needed

File: install_debs_into_image.sh (hunk 1 above)
Comment (iluminat23) [reply install_debs_into_image.sh:69]: And if this is an issue, then we need to do this in our customize_image script as well?

File: install_debs_into_image.sh (hunk 2 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:186]: agree, hopefully the image building can be done in CI as soon as possible.

File: install_debs_into_image.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply install_debs_into_image.sh:5]: get it, agree

File: install_debs_into_image.sh (hunk 3 above)
Comment (zhan-kunbus) [reply install_debs_into_image.sh:169]: I would think this is more temporary than other changes, as normally files will be delivered only with debian packages.

**Comment by zhan-kunbus** [comment]
To sum up, the changes in this PR are not needed to release to customer. It should be enough to put it in devel branch and be used to generate a developing image.

File: install_debs_into_image.sh (hunk 1 above)
Comment (iluminat23) [reply install_debs_into_image.sh:69]: I need to object on both statements.
I don't see why this behavior might be triggered be any operation before. We do a umount, which should take care of everything.
A sync command can impact the performance of the whole system. As **all** filesystems will be synced. And thus might heavily affect the performance of a build system with multiple tasks running. Even running this in a VM can impact the performance of the host system.

Comment (zhan-kunbus) [reply install_debs_into_image.sh:69]: One more information, when the dirty bit occurs, the target image is located in a different mount point from where imagebakery located, that it, the imagebakery located in /dev/sda8/   and target image in /dev/sdb1
However, I tried to reproduce it with the environment above, but it doesn't happen anymore.
